arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,368 @@
+        }
+
+        // Rule 1: The vending machine will distribute a cupcake to anyone who hasn't recently received one.
+        uint fiveSeconds = 5 * 1 seconds;
+        bool userCanReceiveCupcake = _cupcakeDistributionTimes[userAddress] + fiveSeconds <= block.timestamp;
+        if (userCanReceiveCupcake) {
+            _cupcakeBalances[userAddress]++;
+            _cupcakeDistributionTimes[userAddress] = block.timestamp;
+            return true;
+        } else {
+            revert("HTTP 429: Too Many Cupcakes (you must wait at least 5 seconds between cupcakes)");
+        }
+    }
+
+    // Getter function for the cupcake balance of a user
+    function getCupcakeBalanceFor(address userAddress) public view returns (uint) {
+        return _cupcakeBalances[userAddress];
+    }
+}
+```
+
+Run `npx hardhat compile` again. You should see `Compiled 1 Solidity file successfully` in the terminal output. You should also see a new `decentralized-cupcakes/artifacts/contracts/VendingMachine.sol` directory.
+
+
+### Deploy the smart contract locally
+
+To deploy our `VendingMachine` smart contract locally, we'll use two terminal windows and a wallet: 
+
+1. We'll use the first terminal window to run Hardhat's built-in local Ethereum node
+2. We'll then configure a wallet so we can interact with our smart contract after it's deployed to (1)
+3. We'll then deploy our smart contract to (1)'s node
+
+
+#### Run a local Ethereum node
+
+// pull explanation up here - emulator etc
+
+Run `npx hardhat node` from your `decentralized-cupcakes` directory to begin running a local Ethereum node. You should see something along the lines of `Started HTTP and WebSocket JSON-RPC server at http://127.0.0.1:8545/` in your terminal. You should also see a number of test accounts automatically generated for you:
+
+```
+Account #0: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 (10000 ETH)
+Private Key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
+
+...etc
+```
+
+<!-- todo: psst, don't ever share your mainnet wallet private keys or even copy them into your clipboard -->
+
+Note that in the context of this quickstart, "account" refers to a public wallet address and its associated private key.
+
+#### Configure Metamask
+
+Next, open Metamask and create or import a wallet by following the displayed instructions. By default, Metamask will connect to Ethereum mainnet. To connect to our local "testnet", enable test networks through Metamask's settings. Then select the `Localhost 8545` network:
+
+![Connect Metamask to Localhost 8545](assets/quickstart-localhost-metamask.png)
+
+Your mainnet wallet won't have a balance on your local testnet's node, but we can import one of the test accounts into Metamask to gain access to 10,000 fake ETH. Copy the private key of one of the test accounts (**excluding** the `0x` prefix, so eg `ac0..f80`) and import it into Metamask:
```

Comment:
Access to the private key is a bit hidden in Metamask, so maybe we should clarify this further => 3-dots / Account details / Export private key.
Also, private key does not have the "0x" prefix in Metamask.

Reply:
This is instructing the developer to import the test account into metamask, from the local node's console output, and then removing the `0x` prefix before importing (procedure verified as of last week, Matamask vLatest). Can you clarify your feedback?

Is this not clear, when reviewing the content in its [rendered preview](https://nitro-docs-git-boop-quickstart-offchain-labs.vercel.app/for-devs/quickstart-solidity-hardhat) form? If not, I'll iterate until it is :).

Reply:
Oh, you're right, sorry about that. I don't know why I thought it was doing the operation the other way around (Metamask => hardhat). It is well explained as it is.